setup.py

```diff
@@ -64,6 +64,7 @@
         "teachable": ["chromadb"],
         "lmm": ["replicate", "pillow"],
         "graph": ["networkx", "matplotlib"],
+        "gemini": ["google-generativeai", "pillow", "pydantic==1.10.9"],
```

Comment:
Why version 1.x of pydantic? It will stop being supported even for security bugs very soon, see https://docs.pydantic.dev/latest/version-policy/#pydantic-v1

### Pydantic V1

Active development of V1 has already stopped, however critical bug fixes and security vulnerabilities will be fixed in V1 for one year after the release of V2 (June 30, 2024).

We should also limit version ranges of other packages, e.g.
```suggestion
        "gemini": ["google-generativeai>=0.5,<1", "pillow", "pydantic==1.10.9"],
```
The new major version will surely break everything and we should make sure that it doesn't get installed at some future point.